lightning/src/util/config.rs

```diff
@@ -397,6 +397,38 @@ pub struct ChannelConfig {
 	/// [`Normal`]: crate::chain::chaininterface::ConfirmationTarget::Normal
 	/// [`Background`]: crate::chain::chaininterface::ConfirmationTarget::Background
 	pub force_close_avoidance_max_fee_satoshis: u64,
+	/// If set, allows this channel's counterparty to skim an additional fee off this node's inbound
+	/// HTLCs. Useful for liquidity providers to offload on-chain channel costs to end users.
+	///
+	/// Usage:
+	/// - The payee will set this option and set its invoice route hints to use [intercept scids]
+	///   generated by this channel's counterparty.
+	/// - The counterparty will get an [`HTLCIntercepted`] event upon payment forward, and call
+	///   [`forward_intercepted_htlc`] with less than the amount provided in
+	///   [`HTLCIntercepted::expected_outbound_amount_msat`]. The difference between the expected and
+	///   actual forward amounts is their fee.
+	// TODO: link to LSP JIT channel invoice generation spec when it's merged
+	///
+	/// # Note
+	/// It's important for payee wallet software to verify that [`PaymentClaimable::amount_msat`] is
+	/// as-expected if this feature is activated, otherwise they may lose money!
+	/// [`PaymentClaimable::counterparty_skimmed_fee_msat`] provides the fee taken by the
+	/// counterparty.
```

Comment:
Not sure if it's worth mentioning here or is maybe more fit for an LSP client spec, but I think the attack described in https://github.com/lightningdevkit/rust-lightning/pull/2319#discussion_r1228771187 is still possible if the payee fails a payment based on the skimmed fee being too high instead of just checking `amount_msat` and that the sum of `amount_msat + counterparty_skimmed_fee_msat` add up to at least what they expected. The docs here already only say to check `amount_msat` so I think it's probably fine as is but figured I'd still make note of it :)

Reply:
Oh my bad, for some reason I was thinking intermediate nodes could still affect the final skimmed fee but that was fixed, oops